applications/vanilla/controllers/class.discussioncontroller.php

```diff
@@ -188,6 +175,18 @@ public function index($DiscussionID = '', $DiscussionStub = '', $Page = '') {
         // Load the comments
         $this->setData('Comments', $this->CommentModel->getByDiscussion($DiscussionID, $Limit, $this->Offset));
 
+        $LatestItem = $this->Discussion->CountCommentWatch;
```

Comment:
Was this done in order to keep the comment-oriented operations together or is there a functional reason?

Reply:
@tburry, can you speak to this?

Reply:
This was done when we thought we were going to add another correction for `$LatestItem` based on the comments that were fetched from the DB. That ended up not being the case, but the move was not undone.

I would say though that the correction (comparing the last page of comments to date last viewed) is sensible and may be done in the future if we can figure a way around the UX issues with it. In that case, leaving the code here is still sensible.